src/commands/registerCommands.ts

```diff
@@ -179,24 +181,86 @@ async function grammarAssociationCommand(documentURI: Uri, languageClient: Langu
   const fileUri = await window.showOpenDialog(options);
   if (fileUri && fileUri[0]) {
     // The XSD, DTD has been selected, get the proper syntax for binding this grammar file in the XML document.
-    const identifier = TextDocumentIdentifier.create(documentURI.toString());
     const grammarURI = fileUri[0];
     try {
-      const result = await commands.executeCommand(ServerCommandConstants.ASSOCIATE_GRAMMAR_INSERT, identifier, grammarURI.toString(), bindingType);
-      // Insert the proper syntax for binding
-      const lspTextDocumentEdit = <TextDocumentEdit>result;
-      const workEdits = new WorkspaceEdit();
-      for (const edit of lspTextDocumentEdit.edits) {
-        workEdits.replace(documentURI, languageClient.protocol2CodeConverter.asRange(edit.range), edit.newText);
+      const currentFile = (window.activeTextEditor && window.activeTextEditor.document && window.activeTextEditor.document.languageId === 'xml') ? window.activeTextEditor.document : undefined;
+      if (bindingType == 'fileAssociation') {
+        // Bind grammar using file association
+        await bindWithFileAssociation(documentURI, grammarURI, currentFile);
+      } else {
+        // Bind grammar using standard binding
+        await bindWithStandard(documentURI, grammarURI, bindingType, languageClient);
       }
-      workspace.applyEdit(workEdits); // apply the edits
-
     } catch (error) {
       window.showErrorMessage('Error during grammar binding: ' + error.message);
     };
   }
 }
 
+/**
+ * Perform grammar binding using file association through settings.json
+ *
+ * @param documentURI the URI of the current XML document
+ * @param grammarURI the URI of the user selected grammar file
+ * @param document the opened TextDocument
+ */
+async function bindWithFileAssociation(documentURI: Uri, grammarURI: Uri, document: TextDocument) {
+  const absoluteGrammarFilePath = grammarURI.toString();
+  const currentFilename = getFileName(documentURI.toString());
+  const currentWorkspaceUri = getAbsoluteWorkspaceUri(document);
```

Comment:
Replace with:

```ts
const currentWorkspaceUri = getWorkspaceUri(document);
  if (!currentWorkspaceUri) {
    return;
  }
  // If the grammar file is in the same workspace, use the relative path, otherwise use the absolute path
  const grammarFilePath = getDirectoryPath(absoluteGrammarFilePath).includes(currentWorkspaceUri.toString()) ? getRelativePath(currentWorkspaceUri.toString(), absoluteGrammarFilePath) : absoluteGrammarFilePath;

```